### ISSUE TITLE
Annotation features with missing information in Genbank files are merged with the previous line in dictionary

### ISSUE THREAD

**Issue by gjaq** [biopython/biopython#4492]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
>>> import sys; print(sys.version)
3.10.8 (main, Oct 11 2022, 11:35:05) [GCC 11.2.0]
>>> import platform; print(platform.python_implementation()); print(platform.platform())
CPython
Linux-5.15.0-88-generic-x86_64-with-glibc2.35
>>> import Bio; print(Bio.__version__)
1.81
```

### Expected behaviour

I am processing a genbank file using SeqIO and extracting information from the annotation. However, some attributes of the genbank file are empty and, in the resulting dictionary, they are merged with the line before.

For example, when trying to extract the information about complete and partial rRNAs using ```seq_record.annotations['structured_comment']['Genome-Annotation-Data']['complete rRNAs']``` and ```seq_record.annotations['structured_comment']['Genome-Annotation-Data']['partial rRNAs']``` from an annotation that looks like this:

```
rRNAs                             :: 2, 1, 1 (5S, 16S, 23S)
complete rRNAs                    :: 2, 1, 1 (5S, 16S, 23S)
partial rRNAs                     ::
tRNAs                             :: 34
```

I would expect to obtain:

```python
>>> print(seq_record.annotations['structured_comment']['Genome-Annotation-Data']['complete rRNAs'])
2, 1, 1 (5S, 16S, 23S)
```

### Actual behaviour

Instead, what I obtain is this: 

```python
>>> print(seq_record.annotations['structured_comment']['Genome-Annotation-Data']['complete rRNAs'])
2, 1, 1 (5S, 16S, 23S) partial rRNAs                     ::
``` 

So when a line contains no information, it gets merged to the values of the previous line. In this case, the key ```partial rRNAs``` does not exist.

### Steps to reproduce

```python
from Bio import SeqIO
input = "GCF_012367175.1_ASM1236717v1_genomic.gbff"
contig = SeqIO.index(input, "genbank")['NZ_JAAEVG010000101.1']
print(contig.annotations['structured_comment']['Genome-Annotation-Data']['complete rRNAs'])
```

[GCF_012367175.1_ASM1236717v1_genomic.gbff.zip](https://github.com/biopython/biopython/files/13365208/GCF_012367175.1_ASM1236717v1_genomic.gbff.zip)

**Comment by peterjc** [comment]
@biologyguy as our GenBank structured_comment expert, would you be able to take at look at this please?

**Comment by biologyguy** [comment]
Sure, I can have a look this weekend. Been out of the loop for awhile though!

**Comment by biologyguy** [comment]
Yep, totally a bug! 30 minutes to squash it, another 3 hours to remind myself how to contribute to BioPython 😄

**Comment by peterjc** [comment]
If this is an "easy fix", you might get away with making the change in the GitHub editor which would guide you though submitting a pull request.

But if you're planning a test case too, doing it at the command line is probably worth the effort. i.e. Update your local repository, make a new branch, commit changes, test, push to your fork, make a pull request. I can understand this being a pain if you don't use git(hub) often.

(If the 3 hour estimate was your way of saying you'd struggle with this, there is the old fashioned approach - email me the changes. e.g. patch file, or just modified versions of the Python code. That way takes more of my time though, but not three hours worth.)

**Comment by biologyguy** [comment]
Ah, no no, it's all good. I shouldn't have blamed biopython so flippantly. I made my life harder by doing the work on my PC instead of my mac (totally messed up my conda for no good reason), forgot that I needed to work off a fork instead of trying to push a branch to biopython itself, used pytest instead of unittest... Just a bunch of cobwebs to shake loose and the child requiring a bit of attention here and there.
It's nice to look at this again. Pretty sure my MR is on the up and up, but let me know if it's not.

**Comment by peterjc** [comment]
I've lost count of how many times I've had to wipe my conda setup and reinstall it :wink:

**Comment by biologyguy** [comment]
@gjaq
Were you able to check out my branch from a few weeks ago and confirm the fix?
@peterjc 
Was there anything else you needed from me on the MR?

**Comment by gjaq** [comment]
@biologyguy yes, it fixed my problem, thank you for the changes!

**Comment by peterjc** [comment]
Merged - thanks all, independent testing of the fix is a bonus.
